### ISSUE TITLE
build: AppVeyor build uses Libevent that built without support for evthread_use_windows_threads()

### ISSUE THREAD

**Issue by hebasto** [bitcoin/bitcoin#17586]
With [this](https://github.com/hebasto/bitcoin/commit/7822f02c061402a07fd08421f33feb5f27c446e5) patch to master:

https://ci.appveyor.com/project/hebasto/bitcoin/builds/29096172

Ref: [EVTHREAD_USE_WINDOWS_THREADS_IMPLEMENTED](http://www.wangafu.net/~nickm/libevent-2.0/doxygen/html/thread_8h.html#a05306512f3074b45be6eba702440f1f1)

**Comment by laanwj** [comment]
Seems that, while an unlikely build configuration, it's still possible to build libevent without thread support:
```
cmake EVENT_CMAKE_OPTIONS="-DEVENT__DISABLE_THREAD_SUPPORT=ON"
```
Though it's curious that the AppVeyor error is a link-time error, not a compile-time error. So apparently in the header it does still define the function signature for `evthread_use_windows_threads`.

**Comment by hebasto** [comment]
> Though it's curious that the AppVeyor error is a link-time error, not a compile-time error.

The latest builds have compile-time errors.

**Comment by sipsorcery** [comment]
I've confirmed I get the same compiler error after installing `libevent` using the latest vcpkg port. The error is caused by the line below in the port file. Commenting that line out and reinstalling the libevent package allows the build to succeed.

````
thread  EVENT__DISABLE_THREAD_SUPPORT
````

One way to fix the build issue is to create a custom libevent port and then perform the vcpkg install using:

````
vcpkg install --triplet x64-windows-static --overlay-ports=..\vcpkg-custom-ports\ports berkeleydb boost-filesystem boost-multi-index boost-signals2 boost-test boost-thread libevent rapidcheck zeromq double-conversion
````

Any ports in the `..\vcpkg-custom-ports\ports` take precedence.

**Comment by laanwj** [comment]
@sipsorcery I think you need to request the "thread" feature explicitly now, somehow

**Comment by hebasto** [comment]
I'm not a msvc-guru, so it is curious if there is a way to specify the package version in `vcpkg install ...`?

**Comment by sipsorcery** [comment]
Wumpus has just enlightened me to vcpkg's [Feature Packages](https://devblogs.microsoft.com/cppblog/vcpkg-introducing-installation-options-with-feature-packages/). Testing that now.

@hebasto no vcpkg does not allow specifying of a specific version. The [Overlay Ports](https://devblogs.microsoft.com/cppblog/vcpkg-2019-06-update/) is a defacto way to do it but it does require additional steps.

**Comment by JackBoosY** [comment]
Hi guys,  thanks for reporting this issue!
I will handle it tomorrow.

**Comment by fanquake** [comment]
Fixed in #17592.
